components/features/article-chat.tsx

```
@@ -0,0 +1,461 @@
+  SelectItem,
+  SelectTrigger,
+} from "@/components/ui/select";
+import useLocalStorage from "@/lib/hooks/use-local-storage";
+import { useArticleChat, UsageData } from "@/lib/hooks/use-chat";
+import { Response } from "../ai/response";
+import { isTextUIPart, UIMessage } from "ai";
+import {
+  ArrowUp,
+  Square,
+  Zap,
+  Infinity,
```

Biome (2.3.13)

[error] Do not shadow the global "Infinity" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)